src/components/layout/header/Search.tsx

```diff
@@ -142,13 +143,25 @@ export function Search() {
   const [OverseerrResults, setOverseerrResults] = useState<any[]>([]);
   const [opened, setOpened] = useState(false);
 
-  useEffect(() => {
-    if (debounced !== '' && selectedSearchEngine.value === 'overseerr' && searchQuery.length > 3) {
-      axios.get(`/api/modules/overseerr?query=${searchQuery}`).then((res) => {
-        setOverseerrResults(res.data.results ?? []);
-      });
+  const { data, isLoading, error } = useQuery(
+    ['overseerr', debounced],
+    async () => {
+      if (debounced !== '' && selectedSearchEngine.value === 'overseerr' && debounced.length > 3) {
```

Comment:
Redundant condition

Reply:
What do you mean redundant ? It's the one we had before

Reply:
``debounced !== '' && denounced.length > 3`` is redundant. See it?

Reply:
> ``debounced !== '' && denounced.length > 3`` is redundant. See it?

Ahhh. It's a logic flaw on my end I didn't notice it. Should be the queried string

Reply:
Acutally it makes sense. Debounced is the debounced input and we don't search before 4 caracters